tests/test_transform.py

```diff
@@ -252,6 +255,65 @@ def test_null_object_transform(self):
         empty_data = {'addrs': {}}
         self.assertDictEqual(empty_data, transform(empty_data, schema))
 
+    def test_decimal_types_transform(self):
+        schema =  {"type": "object",
+                   "properties": {"percentage": {"type": ["null", "string"],
```

Comment:
These cases look good overall, I'd like to see them all with just `"type": "string"` to catch the nullability issue I mentioned above. Some of them should fail validation (correctly) in that case.